### PR TITLE
Remove provider aws

### PR DESCRIPTION
**Pull request by landerss1**
This PR removes support for the AWS cloud provider.

### CODE REVIEW

File: modules/kubernetes/ingress-nginx/templates/values.yaml.tpl [hunk 1]
```diff
@@ -25,23 +25,12 @@ controller:
...
-  # Optionally change this to ClusterFirstWithHostNet in case you have 'hostNetwork: true'.
-  # By default, while using host network, name resolution uses the host's DNS. If you wish nginx-controller
-  # to keep resolving names inside the k8s network, use ClusterFirstWithHostNet.
-  dnsPolicy: ClusterFirstWithHostNet
-  hostNetwork: true
-  %{~ endif ~}
-
   service:
     externalTrafficPolicy: Local
-    %{~ if provider == "aws" || internal_load_balancer || external_dns_hostname != "" ~}
```
Comment (phillebaba) [inline modules/kubernetes/ingress-nginx/templates/values.yaml.tpl:38]: We can keep this but I am not sure if it is beeing used in Azure.

File: modules/kubernetes/cluster-autoscaler/templates/values.yaml.tpl [hunk 2]
```diff
@@ -1,4 +1,4 @@
-cloudProvider: ${provider}
```
Comment (phillebaba) [inline modules/kubernetes/cluster-autoscaler/templates/values.yaml.tpl:1]: Cluster autoscaler module should be completely removed as it is not used in Azure. AKS runs this for you.

**Review by phillebaba — COMMENTED** [review]
End of an era. Things looks good and we are bound to have missed

**Review by CalleB3 — APPROVED** [review]

File: modules/kubernetes/ingress-nginx/templates/values.yaml.tpl (hunk 1 above)
Comment (landerss1) [reply modules/kubernetes/ingress-nginx/templates/values.yaml.tpl:38]: Just removed the aws condition, so should keep working.

File: modules/kubernetes/cluster-autoscaler/templates/values.yaml.tpl [hunk 3]
```diff
@@ -1,4 +1,4 @@
...
+cloudProvider: "azure"
```
[the remark this answers is not in the document]
Comment (landerss1) [reply modules/kubernetes/cluster-autoscaler/templates/values.yaml.tpl:1]: Done.

File: modules/kubernetes/cluster-autoscaler/templates/values.yaml.tpl (hunk 2 above)
Comment (landerss1) [reply modules/kubernetes/cluster-autoscaler/templates/values.yaml.tpl:1]: Done, History.

**Review by phillebaba — APPROVED** [review]
LGTM

**Pull request merged**